src/fa.h

```diff
@@ -25,10 +25,15 @@
 
 #include <stdio.h>
 #include <regex.h>
+#include <stdbool.h>
+#include <stdint.h>
 
 /* The type for a finite automaton. */
 struct fa;
 
+/* The type of a state of a finite automaton */
+struct state;
```

Comment:
This should carry a dire warning that pointers to struct state are only valid across calls to `fa_state_*`. If you get a `struct state *` and call any other `fa_*` function, you must consider the pointer invalid (since we could have garbage collected the state as a result of that call)

Reply:
I don't really know what you mean. Calling any **not** `fa_state*` function with a `struct state *` will cause an error at compilation time.